<!-- Demo within Gazebo to demonstrate peg in hole task on top of a rover -->
<launch>

  <!-- Use simulated time -->
  <param name="use_sim_time" value="true"/>

  <!-- Make the robot model discoverable by Gazebo -->
  <env name="IGN_GAZEBO_RESOURCE_PATH" value="$(find phantomx_pincher_description)"/>

  <!-- Start the MoveIt setup in Gazebo simulation without spawning the robot -->
  <include file="$(find phantomx_pincher)/launch/gz.launch">
    <arg name="world" value="$(find phantomx_pincher_demos)/worlds/rover_peg_in_hole.sdf"/>
    <arg name="render_engine" value="ogre"/>
    <arg name="spawn_robot" value="false"/>
    <arg name="start_delay" value="1.0"/>
    <arg name="rviz_config" default="$(find phantomx_pincher_demos)/rviz/demo_rover_peg_in_hole.rviz"/>
  </include>

  <!-- ros_ign_bridge (point cloud -> ROS) -->
  <node name="$(anon ros_ign_bridge)" pkg="ros_ign_bridge" type="parameter_bridge" args="/rgbd_camera/points@sensor_msgs/PointCloud2[ignition.msgs.PointCloudPacked"/>

  <!-- Node that demonstrates a peg in hole routine -->
  <node name="demo_peg_in_hole" pkg="phantomx_pincher_demos" type="demo_peg_in_hole.py"/>

</launch>
